### PR TITLE
refactor: use swappers in fee manager

### PR DESCRIPTION
**Pull request by 0xsambugs**
This PR is pretty cool. We are:
1. Making the fee manager use the `RunSwap` extensions
2. This allows us to:
    * Avoid the special case of ETH/WETH and remove related code
    * Add swap capabilities to the fee manager
    * Re-use a lot of the code from the `SwapAdapter`
3. We now also have the first integration tests that uses everything in our new architecture
    * The DCAHub
    * The DCAFeeManager
    * The DCASwapper
    * The TransformerRegistry
    * The SwapperRegistry
    * The SwapAdapter

### CODE REVIEW

**Review by nchamo — APPROVED** [review]
I love that each of your PRs deletes more code than it adds ❤️

**Review by 0xged — APPROVED** [review]
image

File: contracts/DCAFeeManager/DCAFeeManager.sol [hunk 1]
```diff
@@ -70,11 +66,6 @@ contract DCAFeeManager is AccessControl, Multicall, IDCAFeeManager {
     _hub.withdrawSwappedMany(_positionSets, _recipient);
   }
 
-  /// @inheritdoc IDCAFeeManager
-  function withdrawProtocolToken(uint256 _amount, address payable _recipient) external onlyRole(ADMIN_ROLE) {
-    _recipient.sendValue(_amount);
-  }
-
```
[the remark this answers is not in the document]
Comment (0xsambugs) [reply contracts/DCAFeeManager/DCAFeeManager.sol:77]: We will be able to withdraw it in `withdrawFromBalance`

File: deploy/005_fee_manager.ts [hunk 2]
```diff
@@ -1,48 +1,22 @@
...
-    default:
-      throw new Error(`Unsupported chain '${hre.network.name}`);
-  }
+  const swapperRegistry = await hre.deployments.get('SwapperRegistry');
 
   await deployThroughDeterministicFactory({
     deployer,
     name: 'DCAFeeManager',
-    salt: 'MF-DCAV2-DCAFeeManager-V2',
+    salt: 'MF-DCAV2-DCAFeeManager-V3',
```
[the remark this answers is not in the document]
Comment (0xsambugs) [reply deploy/005_fee_manager.ts:14]: 🤷

**Pull request merged**